src/actions/github.ts

```diff
@@ -0,0 +1,63 @@
+export const parseOAuthCode = () => {
+    const windowUrl = window.location.search;
+    const params = new URLSearchParams(windowUrl);
+
+    const code = params.get("code");
+
+    return code
+}

```

Comment:
i lean more towards SRP functions and adding unit test for them rather than adding all functions in one file

Reply:
I love this, but it will probably require me spending more time on this to understand the best approach for the refactor. I might be overthinking it, but I'm trying to fit the code into the current scaffold.

Let's backlog this since its now added to my personal action items :)